contracts/Removal.sol

```diff
@@ -90,18 +92,26 @@ contract Removal is
     __ERC1155Supply_init_unchained();
     __AccessControl_init_unchained();
     __AccessControlEnumerable_init_unchained();
-    _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
-    _setupRole(PAUSER_ROLE, _msgSender());
-    _setupRole(MINTER_ROLE, _msgSender());
-    tokenIdCounter = 0;
-    name = "Removal";
+    _grantRole(DEFAULT_ADMIN_ROLE, _msgSender());
+    _grantRole(PAUSER_ROLE, _msgSender());
+    _grantRole(MINTER_ROLE, _msgSender());
+    _grantRole(RELEASER_ROLE, _msgSender());
   }
 
-  function registerRestrictedNORIAddress(address restrictedNORIAddress)
-    external
-    onlyRole(DEFAULT_ADMIN_ROLE)
-  {
-    _restrictedNori = RestrictedNORI(restrictedNORIAddress);
+  function registerContractAddresses(
+    RestrictedNORI restrictedNoriAddress_,
+    FIFOMarket marketAddress_
+  ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    _restrictedNori = RestrictedNORI(restrictedNoriAddress_);
+    _market = marketAddress_;
```

Comment:
do we not need to cast this the way `RestrictedNORI` is above?

Reply:
nope!